src/user.rs

```diff
@@ -172,38 +181,58 @@ impl UserOps for IronOxide {
         let mut rt = Runtime::new().unwrap();
         let device_create_options = device_create_options.clone();
 
-        rt.block_on(user_api::generate_device_key(
-            &recrypt,
-            &jwt.try_into()?,
-            password.try_into()?,
-            device_create_options.device_name,
-            &std::time::SystemTime::now().into(),
-            *OUR_REQUEST,
-        ))
+        rt.block_on(
+            user_api::generate_device_key(
+                &recrypt,
+                &jwt.try_into()?,
+                password.try_into()?,
+                device_create_options.device_name,
+                &std::time::SystemTime::now().into(),
+                &OUR_REQUEST,
+            )
+            .boxed()
+            .compat(),
+        )
     }
 
     fn user_delete_device(&self, device_id: Option<&DeviceId>) -> Result<DeviceId> {
         let mut rt = Runtime::new().unwrap();
-        rt.block_on(user_api::device_delete(self.device.auth(), device_id))
+        rt.block_on(
+            user_api::device_delete(self.device.auth(), device_id)
+                .boxed_local() // required because something isn't Send; perhaps related to IronCoreRequest
+                .compat(),
+        )
     }
 
     fn user_verify(jwt: &str) -> Result<Option<UserResult>> {
         let mut rt = Runtime::new().unwrap();
-        rt.block_on(user_api::user_verify(jwt.try_into()?, *OUR_REQUEST))
+        rt.block_on(
+            user_api::user_verify(jwt.try_into()?, *OUR_REQUEST)
+                .boxed()
+                .compat(),
+        )
     }
 
     fn user_get_public_key(&self, users: &[UserId]) -> Result<HashMap<UserId, PublicKey>> {
         let mut rt = Runtime::new().unwrap();
-        rt.block_on(user_api::user_key_list(self.device.auth(), &users.to_vec()))
+        rt.block_on(
+            user_api::user_key_list(self.device.auth(), &users.to_vec())
+                .boxed()
+                .compat(),
+        )
     }
 
     fn user_rotate_private_key(&self, password: &str) -> Result<UserUpdatePrivateKeyResult> {
         let mut rt = Runtime::new().unwrap();
-        rt.block_on(user_api::user_rotate_private_key(
-            &self.recrypt,
-            password.try_into()?,
-            self.device().auth(),
-        ))
+        rt.block_on(
```

Comment:
Did you consider going to the new tokio runtime and bringing all the 0.1 futures into 0.3 to run them instead of doing all this work (which we'll immediately have to undo).

Reply:
I did consider it, but this seemed more incremental (and worked!). I think I would need the unlreased `tokio-compat` to run v1 futures on new Tokio, possibly. This way the tokio work can be in its own PR (and ticket, probably).